metaflow/plugins/tag_cli.py

```diff
@@ -0,0 +1,531 @@
+            lines.append(formatter.format(*line_values))
+        lines.append("")
+        obj.echo("\n".join(lines), err=False)
+    else:
+        # In quiet mode, we display things much more simply so it is parseable.
+        # When displaying by tag, we output twp columns:
+        #  - the tag (user:<tag> or sys:<tag>)
+        #  - the comma separated list of runs
+        # Columns are separated by a semicolon surrounded by a space on each side
+        # to make it consistent with the by_run listing
+        obj.echo_always(
+            " ; ".join(
+                [
+                    "system:%s" % group if is_system else "user:%s" % group,
+                    ",".join(runs),
+                ]
+            )
+        )
+
+
+def _print_tags_for_one_run(obj, run):
+    system_tags = {run.pathspec: run.system_tags}
+    all_tags = {run.pathspec: run.tags}
+    return _print_tags_for_runs_by_groups(obj, system_tags, all_tags, by_tag=False)
+
+
+def _is_prod_token(token):
+    return token and token.startswith("production:")
+
+
+def _get_client_run_obj(obj, run_id, user_namespace):
+    if _is_prod_token(user_namespace):
+        raise CommandException(
+            "Modifying a scheduled run's tags is currently not allowed."
+        )
+
+    flow_name = obj.flow.name
+
+    # handle error messaging for two cases
+    # 1. our user tries to tag a new flow before it is run
+    # 2. our user makes a typo in --namespace
+    try:
+        namespace(user_namespace)
+        Flow(pathspec=flow_name)
+    except MetaflowNotFound:
+        raise CommandException(
+            "No execution found for *%s*. Please run the flow before tagging."
+            % flow_name
+        )
+
+    except MetaflowNamespaceMismatch:
+        raise CommandException(
+            "No execution found for *%s* in namespace *%s*."
+            % (flow_name, user_namespace)
+        )
+
+    # throw an error with message to include latest run-id when run_id is None
+    if run_id is None:
+        latest_run_id = Flow(pathspec=flow_name).latest_run.id
+        msg = (
+            "Please specify a run-id using --run-id.\n"
+            "*%s*'s latest execution in namespace *%s* has id *%s*."
+            % (flow_name, user_namespace, latest_run_id)
+        )
+        raise CommandException(msg)
+    run_id_parts = run_id.split("/")
+    if len(run_id_parts) == 1:
+        run_name = "%s/%s" % (flow_name, run_id)
+    else:
+        raise CommandException("Run-id *%s* is not a valid run-id" % run_id)
+
+    # handle error messaging for three cases
+    # 1. our user makes a typo in --run-id
+    # 2. our user tries to mutate tags on a meson/production run
+    # 3. our user's --run-id does not exist in the default/specified namespace
+    try:
+        namespace(user_namespace)
+        run = Run(pathspec=run_name)
+    except MetaflowNotFound:
+        raise CommandException("No run *%s* found" % run_name)
```

Comment:
To make it consistent with other error messages - `No run *<run>* found for *<flow>* `